Sources/Commands/Utilities/TestingSupport.swift

```diff
@@ -32,21 +32,34 @@ enum TestingSupport {
     ///
     /// - Returns: Path to XCTestHelper tool.
     static func xctestHelperPath(swiftTool: SwiftTool) throws -> AbsolutePath {
-        let xctestHelperBin = "swiftpm-xctest-helper"
-        let binDirectory = try AbsolutePath(validating: CommandLine.arguments.first!,
-            relativeTo: swiftTool.originalWorkingDirectory).parentDirectory
-        // XCTestHelper tool is installed in libexec.
-        let maybePath = binDirectory.parentDirectory.appending(components: "libexec", "swift", "pm", xctestHelperBin)
-        if swiftTool.fileSystem.isFile(maybePath) {
-            return maybePath
+        var triedPaths = [AbsolutePath]()
+
+        func findXCTestHelper(swiftBuildPath: AbsolutePath) -> AbsolutePath? {
+            // XCTestHelper tool is installed in libexec.
+            let maybePath = swiftBuildPath.parentDirectory.parentDirectory.appending(components: "libexec", "swift", "pm", "swiftpm-xctest-helper")
+            if swiftTool.fileSystem.isFile(maybePath) {
+                return maybePath
+            } else {
+                triedPaths.append(maybePath)
+                return nil
+            }
         }
-        // This will be true during swiftpm development.
-        // FIXME: Factor all of the development-time resource location stuff into a common place.
-        let path = binDirectory.appending(component: xctestHelperBin)
-        if swiftTool.fileSystem.isFile(path) {
-            return path
+
+        if let firstCLIArgument = CommandLine.arguments.first {
+            let runningSwiftBuildPath = try AbsolutePath(validating: firstCLIArgument, relativeTo: swiftTool.originalWorkingDirectory)
+            if let xctestHelperPath = findXCTestHelper(swiftBuildPath: runningSwiftBuildPath) {
+                return xctestHelperPath
+            }
         }
-        throw InternalError("XCTestHelper binary not found.")
+
+        // This will be true during swiftpm development or when using swift.org toolchains.
+        let xcodePath = try TSCBasic.Process.checkNonZeroExit(args: "/usr/bin/xcode-select", "--print-path").spm_chomp()
+        let installedSwiftBuildPath = try TSCBasic.Process.checkNonZeroExit(args: "/usr/bin/xcrun", "--find", "swift-build", environment: ["DEVELOPER_DIR": xcodePath]).spm_chomp()
```

Comment:
I added the explicit `DEVELOPER_DIR` override here since I was seeing some confusing behavior from pure `xcrun` in the development scenario. I was basically ending up with the same path as `runningSwiftBuildPath` above. Looking up the Xcode path separately works, though.